pkg/client/cache/listers.go

```diff
@@ -111,6 +111,44 @@ func (s storePodsNamespacer) List(selector labels.Selector) (api.PodList, error)
 	return pods, nil
 }
 
+// Please note that selector is filtering among the pods that have gotten into
+// the store; there may have been some filtering that already happened before
+// that.
+func (s storePodsNamespacer) ListWithPredicate(pickPod func(*api.Pod) bool) (api.PodList, error) {
```

Comment:
@deads2k no, it's not using the SharedInformer. I need to find two kinds of Pods: 1. those with matching labels, ***or*** 2. those who have the [controllerRef ](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L2798) pointing to the RC. Ideally, I should create an index for labels, and another index for controllerRef, and I can find the pods I want by querying both index and merge the results. However I found it hard to make another index for the controllerRef. That's worth another PR.

I cannot delegate List, because that will filter the pods that having matching controllerRef but mismatching labels.

> I would have created an index based on the label/field selector that was previously used for the GC watch, then queried my index

Do we already have any indexer based on the label yet? I search in pkg/controllers, it seems we only index namespace or name.